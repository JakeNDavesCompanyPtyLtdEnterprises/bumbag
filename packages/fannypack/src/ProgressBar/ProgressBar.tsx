import * as React from 'react';
import * as PropTypes from 'prop-types';
import { BoxProps as ReakitBoxProps } from 'reakit/ts';
// @ts-ignore
import _get from 'lodash/get';

import _ProgressBar, { ProgressBarIndicator } from './styled';

export type LocalProgressBarProps = {
  palette?: string;
};
export type ProgressBarProps = ReakitBoxProps & LocalProgressBarProps;

export const ProgressBar: React.FunctionComponent<LocalProgressBarProps> = ({ ...props }) => (
  <_ProgressBar {...props}>
    <ProgressBarIndicator palette={props.palette} value={props.value} />
  </_ProgressBar>
);

ProgressBar.propTypes = {
  palette: PropTypes.string
};

ProgressBar.defaultProps = {
  palette: 'primary'
};

// @ts-ignore
const C: React.FunctionComponent<ProgressBarProps> = ProgressBar;
export default C;
