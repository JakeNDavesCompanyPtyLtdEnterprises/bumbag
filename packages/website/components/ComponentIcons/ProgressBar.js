import React from 'react';

const SvgComponent = props => (
  <svg width="120" height="12" viewBox="0 0 120 12" fill="none" xmlns="http://www.w3.org/2000/svg">
    <rect width="120" height="12" rx="5" fill="white" />
    <path d="M0 5C0 2.23858 2.23858 0 5 0H86.1538V12H5C2.23858 12 0 9.76142 0 7V5Z" fill="#3926A5" />
  </svg>
);

export default SvgComponent;
