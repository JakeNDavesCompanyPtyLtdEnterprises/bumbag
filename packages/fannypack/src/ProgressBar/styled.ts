import { Box } from '../primitives';
import styled, { css, palette, theme } from '../styled';
import { LocalProgressBarProps } from './ProgressBar';

const sizeAttributes: any = {
  medium: css``,
  large: css``
};

export const ProgressBarIndicator = styled(Box)`
  height: 1rem;
  background-color: ${palette()};
  width: ${props => props.value || '0'}%;
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
`;

export const ProgressBar = styled(Box)<LocalProgressBarProps>`
  /* rounded: border-radius: 1rem; */
  width: 100%;
  height: 1rem;
  background-color: ${props => palette(`${props.palette}Tint`)};

  ${props => props.size && sizeAttributes[props.size]}

  & {
    ${theme('fannypack.Badge.base')};
  }
`;

export default ProgressBar;
