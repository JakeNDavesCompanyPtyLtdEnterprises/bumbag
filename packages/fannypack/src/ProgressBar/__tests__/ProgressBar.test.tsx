import * as React from 'react';
import render from '../../_utils/tests/render';
import ProgressBar from '../ProgressBar';

it('renders correctly', () => {
  const { container } = render(<ProgressBar value={50} />);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly when a size prop is set', () => {
  const { container } = render(<ProgressBar size="large" value={50} />);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly when a palette prop is set', () => {
  const { container } = render(<ProgressBar color="primary" value={50} />);
  expect(container.firstChild).toMatchSnapshot();
});
