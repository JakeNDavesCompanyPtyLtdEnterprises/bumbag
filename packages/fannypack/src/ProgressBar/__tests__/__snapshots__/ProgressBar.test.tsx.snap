// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  color: #444bc9 !important;
}

.c1:focus:not(:focus-visible) {
  outline: none;
}

.c2 {
  height: 100%;
  background-color: #444bc9;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  width: 50%;
}

.c0 {
  border-radius: 3px;
  overflow: hidden;
  width: 100%;
  height: 1rem;
  background-color: #ecedf9;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="50"
  class="c0 c1"
  color="primary"
  role="progressbar"
>
  <div
    class="c2 c1"
    color="primary"
    value="50"
  />
</div>
`;

exports[`renders correctly when a palette prop is set 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  color: #444bc9 !important;
}

.c1:focus:not(:focus-visible) {
  outline: none;
}

.c2 {
  height: 100%;
  background-color: #444bc9;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  width: 50%;
}

.c0 {
  border-radius: 3px;
  overflow: hidden;
  width: 100%;
  height: 1rem;
  background-color: #ecedf9;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="50"
  class="c0 c1"
  color="primary"
  role="progressbar"
>
  <div
    class="c2 c1"
    color="primary"
    value="50"
  />
</div>
`;

exports[`renders correctly when a size prop is set 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  color: #444bc9 !important;
}

.c1:focus:not(:focus-visible) {
  outline: none;
}

.c2 {
  height: 100%;
  background-color: #444bc9;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  width: 50%;
}

.c0 {
  border-radius: 3px;
  overflow: hidden;
  width: 100%;
  height: 1rem;
  background-color: #ecedf9;
  height: 2rem;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="50"
  class="c0 c1"
  color="primary"
  role="progressbar"
  style="size: large;"
>
  <div
    class="c2 c1"
    color="primary"
    value="50"
  />
</div>
`;
